cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamHandler.java

```diff
@@ -182,17 +181,16 @@ public void getInfo(HttpRequest request, HttpResponder responder,
   public void create(HttpRequest request, HttpResponder responder,
                      @PathParam("namespace-id") String namespaceId,
                      @PathParam("stream") String stream) throws Exception {
-
-    String accountID = getAuthenticatedAccountId(request);
-    Id.Stream streamId = Id.Stream.from(accountID, stream);
-
     // Verify stream name
-    if (!isValidName(stream)) {
+    if (!Id.isId(stream)) {
```

Comment:
ok i see. I think it would be better to catch the exception from Id.Stream.from() so that it is more clear. Also, that way you we are not checking twice, and are not assuming that Id.Stream.from() calls Id.isId()